### PR TITLE
Ft/codeforafrica get in touch

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

This PR implements the get-in-touch block.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/lib/data/blockify/getInTouch.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { actionFromActionButton } from "@/codeforafrica/lib/data/utils";
```
Comment (koechkevin) [inline apps/codeforafrica/src/lib/data/blockify/getInTouch.js:1]: @kelvinkipruto do we really need this file? I think we only need to rename GetInTouch block field as action.

File: apps/codeforafrica/src/payload/blocks/GetInTouch.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      type: "text",
+    },
+    {
+      name: "subtitle",
+      label: "Sub Title",
+      required: true,
+      type: "text",
+    },
+    {
+      name: "actionButton",
```
Comment (koechkevin) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:21]:
```suggestion
      name: "action",
```

**Review by koechkevin — COMMENTED** [review]
@kelvinkipruto Update Review App cms

**Review by koechkevin — APPROVED** [review]

File: apps/codeforafrica/src/lib/data/rest/index.js [hunk 3]
```diff
@@ -36,6 +36,13 @@ export const api = {
 
 export async function getPageStaticProps(context) {
   const props = await getPageProps(api, context);
+  if (!props) {
+    return {
+      props: {
+        title: "Page not found",
+      },
+    };
+  }
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/rest/index.js:45]: If we're trying to provide default values for 404 page, we can do that in the 404 page. Other pages should just set [`notFound`](https://nextjs.org/docs/pages/api-reference/functions/get-static-props#notfound).

File: apps/codeforafrica/src/payload/blocks/GetInTouch.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  fields: [
+    {
+      name: "title",
+      label: "Title",
+      required: true,
+      type: "text",
+    },
+    {
+      name: "subtitle",
+      label: "Sub Title",
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:16]:
```suggestion
      label: "Subtitle",
```

File: apps/codeforafrica/src/payload/blocks/GetInTouch.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      required: true,
+      type: "text",
+    },
+    {
+      name: "action",
+      label: "Action Button",
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:22]:
```suggestion
      label: "Action",
```

File: apps/codeforafrica/src/payload/blocks/GetInTouch.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      type: "group",
+      fields: [
+        {
+          name: "label",
+          label: "Label",
+          type: "text",
+          required: true,
+        },
+        link({}),
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:31]: Doesn't a `link` field include a `label`?

File: apps/codeforafrica/src/payload/blocks/GetInTouch.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      label: "Subtitle",
+      required: true,
+      type: "text",
+    },
+    {
+      name: "action",
+      label: "Action",
+      type: "group",
+      fields: [link()],
+    },
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:25]: What structure do we get here (JSON) @kelvinkipruto ?

Comment (kelvinkipruto) [reply apps/codeforafrica/src/payload/blocks/GetInTouch.js:25]: @kilemensi The JSON Response looks like this:
```
{
  "action": {
    "label": "GET IN TOUCH",
    "linkType": "internal",
    "doc": {...},
    "href": "/contact"
  }
}
```

Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/GetInTouch.js:25]:
```suggestion
    linkGroup({ overrides: { name: "action", label: "Action" } }),
```

I think we have `linkGroup` field for this.

**Review by kilemensi — APPROVED** [review]

**Pull request merged**